packages/compiler/server/serverlib.ts

```diff
@@ -670,8 +699,20 @@ export function createServer(host: ServerHost): Server {
     }
 
     const references: IdentifierNode[] = [];
-    for (const script of program.sourceFiles.values() ?? []) {
-      visitChildren(script, function visit(node) {
+    if (wholeProgram) {
+      for (const script of program.sourceFiles.values() ?? []) {
+        addReferenceIdentifiers(program, script, sym, references);
+      }
+    } else {
+      addReferenceIdentifiers(program, file, sym, references);
+    }
+    function addReferenceIdentifiers(
+      program: Program,
+      file: CadlScriptNode,
```

Comment:
I think the only arg you need to pass here is the `file` all the others are already accesible in the function or move this function outside

Reply:
Yes, I also think sym is needed since it can be undefined

Reply:
well sym is also computed before and the same in both cases so should be able to omit it as well